rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -670,6 +673,46 @@ public Topology getTopology(@HeaderParam("sessionid") String sessionId) throws N
         return rm.getSupportedNodeSourceInfrastructures();
     }
 
+    @Override
+    @GET
+    @GZIP
+    @Path("infrastructuresmapping")
+    @Produces("application/json")
+    public Map<String, List<String>> getInfrasToPoliciesMapping(@HeaderParam("sessionid") String sessionId)
+            throws NotConnectedException {
+        ResourceManager rm = checkAccess(sessionId);
+        Map<String, List<String>> result = rm.getInfrasToPoliciesMapping();
+        Collection<PluginDescriptor> supportedInfrastructures = rm.getSupportedNodeSourceInfrastructures();
+        Collection<PluginDescriptor> supportedPolicies = rm.getSupportedNodeSourcePolicies();
+        Iterator<Map.Entry<String, List<String>>> iterator = result.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<String, List<String>> entry = iterator.next();
+            boolean found = false;
+            for (PluginDescriptor supportedInfrastructure : supportedInfrastructures) {
+                if (supportedInfrastructure.getPluginName().equals(entry.getKey())) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                iterator.remove();
+            }
+        }
+        return result.entrySet().stream().map(entry -> {
+            List<String> policies = entry.getValue().stream().filter(policy -> {
+                for (PluginDescriptor supportedPolicy : supportedPolicies) {
+                    if (supportedPolicy.getPluginName().equals(policy)) {
+                        return true;
+                    }
+                }
+                return false;
+            }).collect(Collectors.toList());
+            ;
+            return new AbstractMap.SimpleEntry<>(entry.getKey(), policies);
```

Comment:
Please extract the content of the result.entrySet().stream().map() call to improve the readibility of your code. This is quiet obfuscated ! (this is a shared statement.)

Reply:
true